models/src/main/scala/it/pagopa/interop/catalogmanagement/model/persistence/CatalogEventsSerde.scala

```diff
@@ -0,0 +1,63 @@
+package it.pagopa.interop.catalogmanagement.model.persistence
+
+import it.pagopa.interop.catalogmanagement.model.persistence.JsonFormats._
+import it.pagopa.interop.commons.queue.message.ProjectableEvent
+import spray.json._
+
+object CatalogEventsSerde {
+
+  val projectableCatalogToJson: PartialFunction[ProjectableEvent, JsValue] = { case event: Event =>
+    catalogToJson(event)
+  }
+
+  def catalogToJson(event: Event): JsValue = event match {
+    case x: CatalogItemAdded                  => x.toJson
+    case x: ClonedCatalogItemAdded            => x.toJson
+    case x: CatalogItemUpdated                => x.toJson
+    case x: CatalogItemWithDescriptorsDeleted => x.toJson
+    case x: CatalogItemDocumentUpdated        => x.toJson
+    case x: CatalogItemDeleted                => x.toJson
+    case x: CatalogItemDocumentAdded          => x.toJson
+    case x: CatalogItemDocumentDeleted        => x.toJson
+    case x: CatalogItemDescriptorAdded        => x.toJson
+    case x: CatalogItemDescriptorUpdated      => x.toJson
+  }
+
+  val jsonToCatalog: PartialFunction[String, JsValue => ProjectableEvent] = {
+    case `catalogItemAdded`                  => _.convertTo[CatalogItemAdded]
+    case `clonedCatalogItemAdded`            => _.convertTo[ClonedCatalogItemAdded]
+    case `catalogItemUpdated`                => _.convertTo[CatalogItemUpdated]
+    case `catalogItemWithDescriptorsDeleted` => _.convertTo[CatalogItemWithDescriptorsDeleted]
+    case `catalogItemDocumentUpdated`        => _.convertTo[CatalogItemDocumentUpdated]
+    case `catalogItemDeleted`                => _.convertTo[CatalogItemDeleted]
+    case `catalogItemDocumentAdded`          => _.convertTo[CatalogItemDocumentAdded]
+    case `catalogItemDocumentDeleted`        => _.convertTo[CatalogItemDocumentDeleted]
+    case `catalogItemDescriptorAdded`        => _.convertTo[CatalogItemDescriptorAdded]
+    case `catalogItemDescriptorUpdated`      => _.convertTo[CatalogItemDescriptorUpdated]
+  }
+
+  def getKind(e: Event): String = e match {
+    case CatalogItemAdded(_)                     => catalogItemAdded
```

Comment:
```suggestion
    case _: CatalogItemAdded                     => catalogItemAdded
```
unapply not needed here

Reply:
LGTM, but I suggest applying the same pattern in all the mgmt (e.g. purpose_manangement use unapply)